### PR TITLE
added the support of query in protostream format (without projection)

### PR DESCRIPTION
**Pull request by NeerajGartia21**

### CODE REVIEW

**Review by rigazilla — CHANGES_REQUESTED** [review]
Overall looks great! I've added a comment on the query return value.
Could you also please add a couple of test with a `where` clause?
- 1 test with a query that return 1 value
- 1 test with a query that return an empty resultSet

File: lib/infinispan.js [hunk 1]
```diff
@@ -226,6 +226,23 @@
         var decoder = p.decodeValue();
         return futureKey(ctx, 0x03, k, p.encodeKey(k), decoder);
       },
+      /**
+       * Query the server with the given queryString.
+       *
+       * @param q {(Object)} query to retrieve.
+       * @returns {module:promise.Promise.<?Object[]>}
+       * A promise that will be completed with the values associated with
+       * the query, or undefined if the no values matches the query.
```
[the remark this answers is not in the document]
Comment (NeerajGartia21) [reply lib/infinispan.js:235]: Yes sir. It will be great!
I will add these changes.

File: spec/protostream_spec.js [hunk 2]
```diff
@@ -150,3 +151,36 @@ describe("Put/Get protostream object to/from Infinispan", function () {
...
+                var payload = { awesomeField: "AwesomeString"+i };
+                var message = AwesomeMessage.create(payload);
+                await client.put(i,message)
+            }
+            var queryResp1 = await client.query({queryString:`from awesomepackage.AwesomeMessage ORDER BY awesome_field`});
+            expect(queryResp1.length).toBe(10);
+            var queryResp2 = await client.query({queryString:`from awesomepackage.AwesomeMessage a where a.awesome_field='AwesomeString1'`});
+            expect(queryResp2.length).toBe(1);
+            var queryResp3 = await client.query({queryString:`from awesomepackage.AwesomeMessage a where a.awesome_field='AwesomeString100'`});
+            expect(JSON.stringify(queryResp3)).toBe(JSON.stringify([]));
```
Comment (rigazilla) [inline spec/protostream_spec.js:174]: why you need JSON?
```suggestion
            expect(queryResp3.length).toEqual(0);
```

File: spec/protostream_spec.js [hunk 3]
```diff
@@ -150,3 +151,36 @@ describe("Put/Get protostream object to/from Infinispan", function () {
...
+            await client.clear();
+            for(let i=0;i<10;i++){
+                var payload = { awesomeField: "AwesomeString"+i };
+                var message = AwesomeMessage.create(payload);
+                await client.put(i,message)
+            }
+            var queryResp1 = await client.query({queryString:`from awesomepackage.AwesomeMessage ORDER BY awesome_field`});
+            expect(queryResp1.length).toBe(10);
+            var queryResp2 = await client.query({queryString:`from awesomepackage.AwesomeMessage a where a.awesome_field='AwesomeString1'`});
+            expect(queryResp2.length).toBe(1);
```
Comment (rigazilla) [inline spec/protostream_spec.js:172]: better to also a little check on the content
```suggestion
            expect(queryResp2.length).toBe(1);
            expect(queryResp2[0].awesomeField).toEqual('AwesomeString1');
```

**Review by rigazilla — COMMENTED** [review]
some minor changes proposed

File: spec/protostream_spec.js (hunk 2 above)
Comment (NeerajGartia21) [reply spec/protostream_spec.js:174]: I added that for checking empty array of object and didn't change it.

**Comment by rigazilla** [comment]
merged, thanks @NeerajGartia21!

**Pull request merged**